### PR TITLE
DAOS-14521 test: fix very_perms.py _real_w

### PR DESCRIPTION
**Pull request by daltonbohning**
Test-tag: verify_perms
Skip-unit-tests: true
Skip-fault-injection-test: true

In _real_w, always write a line that allows the file to be executable since opening in "w" mode will truncate the file.

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/util/verify_perms.py [hunk 1]
```diff
@@ -260,12 +260,10 @@ def _real_w(entry_type, path):
     '''
     if entry_type == 'file':
         try:
-            # Write a newline to reduce likelihood of corrupting an executable file
+            # Always write a line that allows the file to be executable
```
Comment (daltonbohning) [inline src/tests/ftest/util/verify_perms.py:263]: It doesn't really matter if we corrupt the file, since this is only for test purposes anyway

**Comment by wiliamhuang** [comment]
@daltonbohning Looks like there are "expected_il_messages" in multiple lines in src/tests/ftest/dfuse/mu_perms.py needed to be doubled other than line 372. 

Line 387, 399, and 414.

**Comment by daltonbohning** [comment]
> @daltonbohning Looks like there are "expected_il_messages" in multiple lines in src/tests/ftest/dfuse/mu_perms.py needed to be doubled other than line 372.
> 
> Line 387, 399, and 414.

Thanks! I need to fix my dev env so I can run this locally again...

**Comment by wiliamhuang** [comment]
"Functional on EL 8" passed.

**Comment by daltonbohning** [comment]
All tests using `verify_perms.py` passed: https://build.hpdd.intel.com/job/daos-stack/job/daos/job/PR-13225/3/testReport/FTEST_dfuse/DfuseMUPerms/

**Review by wiliamhuang — APPROVED** [review]

**Review by phender — APPROVED** [review]

**Pull request merged**